.changeset/lovely-frogs-scream.md

```diff
@@ -0,0 +1,19 @@
+---
+"effect": minor
+---
+
+add Effect.functionWithSpan
+
+Allows you to define an effectful function that is wrapped with a span.
+
+```ts
+import { Effect } from "effect"
+
+const getTodo = Effect.functionWithSpan({
```

Comment:
why not `functionWithSpan(body, options)` ?

Reply:
I tried that first, but it doesn't format very well with dprint.